docs/developing/autofocus.md

```diff
@@ -0,0 +1,29 @@
+---
+title: Autofocus
+---
+
+<head>
+  <title>Autofocus</title>
+  <meta
+    name="description"
+    content="Learn how to manage focus in Ionic applications using the setFocus API instead of the autofocus attribute."
+  />
+</head>
+
+The `autofocus` attribute is a standard HTML attribute that allows developers to set focus to an element when a page loads. This attribute is commonly used to set focus to the first input element on a page. However, the `autofocus` attribute can cause issues in routing applications when navigating between pages. This is because the `autofocus` attribute will set focus to the element when the page loads, but will not set focus to the element when the page is revisited.
+
+# Managing Focus
+
+Ionic provides a `setFocus` API that allows developers to manually set focus to an element. This API should be used in place of the `autofocus` attribute and called within the `ionViewDidEnter` lifecycle event for routing applications, within the `didPresent` lifecycle event for overlays, and the `appload` event for vanilla JavaScript applications.
```

Comment:
We need to also note some of the platform-level restrictions since `setFocus` won't always work either. We don't need to create a detailed list of every little detail, but I think we could stick to the following high level points:

1. Interactive elements can only be focused as a result of a user gesture in Mobile Safari. (In other words, this playground won't work in mobile Safari). So for example, you can call `setFocus` in the `click` callback on a button as long as the user is the one tapping the button. However, you can't programmatically dispatch a click on the button and have `setFocus` work.
2. Android (both mobile Chrome and WebView) require user interaction on the app before `setFocus` can work. This can be as simple as tapping the screen.